nx-X11/programs/Xserver/os/connection.c

```diff
@@ -426,15 +426,26 @@ CreateWellKnownSockets(void)
     if (NoListenAll) {
 	ListenTransCount = 0;
     }
+#ifndef NXAGENT_SERVER
     else if ((displayfd < 0) || explicit_display) {
+#else
+    else if ((displayfd < 0) && explicit_display) {
+#endif /* ! NXAGENT_SERVER */
         if (TryCreateSocket(atoi(display), &partial) &&
```

Comment:
The problem I see is that if `-displayfd` is not given and no explicit display has been selected, the code will jump to the else-branch and do display number autodetection, which breaks the previous behavior (that defaulted to trying to open up display `0` once and fail if it couldn't.)

The condition should thus probably rather be something like

```C
    else if (displayfd < 0) {
```

since if we don't have an displayfd, we can't do autodetection (or rather not write out the autodected display number anywhere.)